### PR TITLE
Core/Spells: Implement SPELL_EFFECT_TELEPORT_TO_RETURN_POINT

### PR DESCRIPTION
**Pull request by matanshukry**
**Changes proposed:**

- Implement SPELL_EFFECT_TELEPORT_TO_RETURN_POINT

**Target branch(es):** 3.3.5/master

- [ ] 3.3.5
- [x] master

**Issues addressed:**

**Tests performed:**

- [x] It builds
- [x] Tested in-game

**Known issues and TODO list:** (add/remove lines as needed)

**Use Case:**
Monk's Zen Pilgrimage spell

### CODE REVIEW

**Comment by matanshukry** [comment]
Looks like 2 builds failed due to the sql file / `create table`.

am I suppose to add a `drop table` statement as well?
Looked at other DDL statements such as `alter table`, they didn't seem to have statements to support running them multiple times.

**Comment by Shauren** [comment]
Build fails for a different reason, you didn't add this table to sql/base/characters_database.sql

**Comment by matanshukry** [comment]
@Aokromes 
https://github.com/TrinityCore/TrinityCore/blob/master/sql/updates/characters/master/2020_12_13_00_characters.sql
This one is missing a drop column statement then?

**Comment by Aokromes** [comment]
> @Aokromes
> [https://github.com/TrinityCore/TrinityCore/blob/master/sql/updates/characters/master/2020_12_13_00_characters.sql](https://github.com/TrinityCore/TrinityCore/blob/master/sql/updates/characters/master/2020_12_13_00_characters.sql?rgh-link-date=2021-03-10T13%3A38%3A47Z)
> This one is missing a drop column statement then?

no, alters doesns't ships drop.

**Comment by illfated** [comment]
Nicely done, the build looks good now.

File: src/server/game/DataStores/DBCEnums.h [hunk 1]
```diff
@@ -1438,7 +1438,8 @@ enum SummonPropFlags
     SUMMON_PROP_FLAG_UNK19           = 0x00040000,
     SUMMON_PROP_FLAG_UNK20           = 0x00080000,
     SUMMON_PROP_FLAG_UNK21           = 0x00100000,          // Totems
-    SUMMON_PROP_FLAG_COMPANION       = 0x00200000
+    SUMMON_PROP_FLAG_COMPANION       = 0x00200000,
+    SUMMON_PROP_FLAG_UNK22           = 0x04000000
```
Comment (Shauren) [inline src/server/game/DataStores/DBCEnums.h:1442]: unrelated?

Comment (matanshukry) [reply src/server/game/DataStores/DBCEnums.h:1442]: yeah it's not related, I just found it while looking for some stuff and I thought it would be better to have it.

Is it fine, or do you prefer it in another PR? or not at all?
